lib/active_admin/inputs/filters/select_input.rb

```diff
@@ -43,6 +43,8 @@ def collection
           else
             super
           end
+        rescue ActiveRecord::QueryCanceled => e
+          raise ActiveRecord::QueryCanceled.new "#{e.message.strip} while querying the values for the :#{method} filter"
```

Comment:
We should include "ActiveAdmin" in the error message for clarity.

```suggestion
        rescue ActiveRecord::QueryCanceled => error
          raise ActiveRecord::QueryCanceled.new "#{error.message.strip} while querying the values for the ActiveAdmin :#{method} filter"
```